src/Analytics/AnomalyDetection/Detectors/BruteForceDetector.php

```
@@ -0,0 +1,291 @@
+            'ip_threshold'              => 10, // Failed attempts from single IP
+            'distributed_threshold'     => 20, // Failed attempts across multiple IPs targeting single user
+            'velocity_threshold'        => 3, // Attempts per minute threshold
+        ];
+    }
+
+    /**
+     * Check for brute force from single IP.
+     *
+     * @param  array<string, mixed>  $data
+     */
+    protected function checkIpBruteForce( string $ip, array $data ): ?Anomaly
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$data'. (undefined)

(UnusedFormalParameter)